### PR TITLE
inf, -inf support for pad

### PR DESCRIPTION
**Pull request by ZenginU**
With this PR we can match with torch API, Also we can use this in the onnx as this will be compatible in constant mode

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/tensor.py                         2               1
------------------------------------------------------------
lines added in the tinygrad folder: 1
```

**Comment by ZenginU** [comment]
Operation was canceled, can you retrigger the CI checks?

**Comment by wozeparrot** [comment]
Oh its cuda again, why is apt install being so slow

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -246,6 +246,7 @@ def flip(self, axis, *args) -> Tensor: return mlops.Flip.apply(self, axis=[x if
   def shrink(self, arg:Tuple[Tuple[int, int], ...]) -> Tensor: return mlops.Shrink.apply(self, arg=arg) if any(x != (0,s) for x,s in zip(arg, self.shape)) else self
   def pad(self, arg: Tuple[Tuple[int, int], ...], value:float=0) -> Tensor:
     ret = mlops.Pad.apply(self, arg=arg) if any(x != (0, 0) for x in arg) else self
+    if isinf(value): return ret + copysign(1,value)*(1/mlops.Pad.apply(Tensor.full(self.shape, value), arg=arg))
```
Comment (wozeparrot) [inline tinygrad/tensor.py:249]: can't you just do `copysign(1, value)/mlops.Pad...` here?

Comment (ZenginU) [reply tinygrad/tensor.py]: Assumed it will be more readable as I was dealing with 1/inf = 0, formatted it now.

**Pull request merged**